src/main/expressions.cc

```diff
@@ -0,0 +1,150 @@
+
+	if ((rc = get_int_property((int*) &entry->op, entry_obj, "op", log)) != AS_NODE_PARAM_OK) {
+		printf("get_int_property failed\n");
+		return rc;
+	}
+
+	if ((rc = get_optional_uint32_property(&entry->count, NULL, entry_obj, "count", log)) != AS_NODE_PARAM_OK) {
+		printf("get_int_property failed\n");
+		return rc;
+	}
+
+	if ((rc = get_optional_uint32_property(&entry->sz, NULL, entry_obj, "sz", log)) != AS_NODE_PARAM_OK) {
+		printf("get_int_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("value").ToLocalChecked()).FromJust()) {
+		rc = get_asval_property(&entry->v.val, entry_obj, "value", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_asval_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("strVal").ToLocalChecked()).FromJust()) {
+		// TODO: Free v.str_val once request is done
+		rc =  get_string_property((char**) &entry->v.str_val, entry_obj, "strVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_string_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("bytesVal").ToLocalChecked()).FromJust()) {
+		// TODO: Free v.bytes_val once request is done
+		rc = get_bytes_property(&entry->v.bytes_val, (int*) &entry->sz, entry_obj, "bytesVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_bytes_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("intVal").ToLocalChecked()).FromJust()) {
+		rc = get_int64_property(&entry->v.int_val, entry_obj, "intVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_int64_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("uintVal").ToLocalChecked()).FromJust()) {
+		rc = get_uint64_property(&entry->v.uint_val, entry_obj, "uintVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_uint64_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("floatVal").ToLocalChecked()).FromJust()) {
+		rc = get_float_property(&entry->v.float_val, entry_obj, "floatVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_float_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("boolVal").ToLocalChecked()).FromJust()) {
+		rc = get_bool_property(&entry->v.bool_val, entry_obj, "boolVal", log);
+		if(rc != AS_NODE_PARAM_OK)
+			printf("get_bool_property failed\n");
+		return rc;
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("ctx").ToLocalChecked()).FromJust()) {
+		entry->v.ctx = NULL;
+		return get_optional_cdt_context(entry->v.ctx, NULL, entry_obj, "ctx", log);
+	}
+
+	if (Nan::Has(entry_obj, Nan::New("listPolicy").ToLocalChecked()).FromJust()) {
+		printf("check listPolicy start");
+		Local<Value> policy_obj = Nan::Get(entry_obj, Nan::New("listPolicy").ToLocalChecked()).ToLocalChecked();
+		
+		if (policy_obj->IsObject() && 
+				get_optional_list_policy(entry->v.list_pol, NULL, policy_obj.As<Object>(), log)) {
+			printf("check listPolicy found");
+			return AS_NODE_PARAM_OK;
+		}
+		printf("check listPolicy failed");
```

Comment:
Should these printf calls be in here?

Reply:
Should be converted into `as_v8_error` statements.